controllers/topology/cluster_controller.go

```diff
@@ -111,6 +116,14 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return ctrl.Result{}, err
 	}
 
+	// Return early, if the Cluster does not use a managed topology.
+	// NOTE: We're already filtering events, but this is a safeguard for cases like e.g. when
+	// there are MachineDeployments which have the topology owned label, but the corresponding
+	// cluster is not topology owned.
+	if cluster.Spec.Topology == nil {
+		return ctrl.Result{}, nil
+	}
```

Comment:
This is the actual fix for the issue, correct? The predicate for the external watch seems more of a safeguard?

Reply:
If users would always run either managed or unmanaged Clusters (where the topology label is set accordingly) either of those would fix the issue.

But that's the fix which will fix it in any case